src/main/kotlin/uk/gov/dluhc/printapi/messaging/service/PrintResponseProcessingService.kt

```diff
@@ -110,13 +110,10 @@ class PrintResponseProcessingService(
 
         certificateRepository.save(certificate)
 
-        sendOnlyFirstMatchingEmail(
-            { certificateNotDeliveredEmailSender.testAndSend(printResponse, certificate) },
-            { certificateFailedToPrintEmailSender.testAndSend(printResponse, certificate) },
-        )
-    }
-
-    private fun sendOnlyFirstMatchingEmail(vararg emailSenders: () -> Boolean) {
-        emailSenders.firstOrNull { it.invoke() }
+        if (printResponse.statusStep == ProcessPrintResponseMessage.StatusStep.NOT_MINUS_DELIVERED) {
+            certificateNotDeliveredEmailSenderService.send(printResponse, certificate)
+        } else if (printResponse.status == ProcessPrintResponseMessage.Status.FAILED) {
```

Comment:
Should there be else if or just 2 ifs? I thought these 2 send emails are not dependant upon each other

Reply:
When we get a PrintResponse with a StatusStep of NOT_MINUS_DELIVERED then its status will be FAILED. 
I didn't think it made sense to send 2 emails for the 1 event hence the `else if`.